server/src/main/java/org/elasticsearch/indices/SystemIndices.java

```diff
@@ -328,6 +331,11 @@ public ExecutorSelector getExecutorSelector() {
      * @return The matching {@link SystemIndexDescriptor} or {@code null} if no descriptor is found
      */
     public @Nullable SystemIndexDescriptor findMatchingDescriptor(String name) {
+        return findMatchingDescriptor(indexDescriptors, name);
+    }
+
+    @Nullable
+    static SystemIndexDescriptor findMatchingDescriptor(SystemIndexDescriptor[] indexDescriptors, String name) {
```

Comment:
I'm not objecting necessarily, but why break this out? I assumed it was for testing in my first pass given it's pkg-private, but there's no tests that use this version looks like.

Reply:
You are right, no need for that to be separate.